src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/TimeOnlyConverter.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class TimeOnlyConverter : JsonConverter<TimeOnly>
+    {
+        private static readonly TimeSpanConverter s_timeSpanConverter = new TimeSpanConverter();
+        private static readonly TimeSpan s_timeOnlyMaxValue = TimeOnly.MaxValue.ToTimeSpan();
+
+        public override TimeOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            TimeSpan timespan = s_timeSpanConverter.Read(ref reader, typeToConvert, options);
```

Comment:
`JsonMetadataServices.TimeSpanCoverter`?

Reply:
`JsonMetadataServices.TimeSpanCoverter` returns `JsonConverter<TimeSpan>` instead of `TimeSpanConverter`, so there's no opportunity for the JIT to devirtualize calls.